src/gpu/metal/SDL_gpu_metal.m

```diff
@@ -4343,6 +4343,10 @@ static void METAL_INTERNAL_DestroyBlitResources(
         MetalRenderer *renderer;
         id<MTLDevice> device = NULL;
 
+        if (debugMode) {
+            SDL_setenv_unsafe("MTL_DEBUG_LAYER", "1", 0);
+        }
```

Comment:
This means that once you create a debug context metal validation will be enabled indefinitely. Should we save and restore the value of this variable?

Reply:
We could also unconditionally set this to 1/0, not sure how the debug layer is supposed to work when using multiple devices though.

Reply:
I ran a few experiments to see how this environment variable behaved when overwriting/unsetting the value, and the results are pretty weird. It seems that once Metal has created a device with `MTL_DEBUG_LAYER=1`, the debug layers will be permanently enabled for the rest of the application's lifespan even if the device is recreated.

I tried calling `SDL_unsetenv_unsafe("MTL_DEBUG_LAYER")` on the variable before device recreation, as well as overwriting its value to "0" with `SDL_setenv_unsafe`. No dice. I checked with `SDL_getenv_unsafe` and the values seemed correct, so it must be the driver itself ignoring the changes.

My best guess is that Metal internally checks for the environment variable somewhere during device creation, and if validation is enabled, then it never re-checks again. Either that or the device is never truly "destroyed" once created -- most Apple machines only have one Device to choose from anyway, so they may keep it alive forever and just reuse it on subsequent Create calls.

So... since it seems like toggling this is moot, I think it's okay to keep it enabled indefinitely. I'm struggling to imagine a real-world scenario where you'd want to create a device in debug mode, and then recreate it in non-debug mode within the same session. The only downside of having the validation layers enabled is some modest driver overhead, which can be entirely avoided by not opting into debug mode.

Reply:
It might be worth leaving a comment with this info so people don't ask this again later. :)

Reply:
Makes sense, added a comment summarizing these findings!